http/transport/tracing_round_tripper.go

```diff
@@ -0,0 +1,51 @@
+// Copyright © 2019 by PACE Telematics GmbH. All rights reserved.
+
+package transport
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/getsentry/sentry-go"
+)
+
+// TracingRoundTripper implements a chainable round tripper for tracing
+type TracingRoundTripper struct {
+	transport http.RoundTripper
+}
+
+// Transport returns the RoundTripper to make HTTP requests
+func (l *TracingRoundTripper) Transport() http.RoundTripper {
+	return l.transport
+}
+
+// SetTransport sets the RoundTripper to make HTTP requests
+func (l *TracingRoundTripper) SetTransport(rt http.RoundTripper) {
+	l.transport = rt
+}
+
+// RoundTrip executes a HTTP request with distributed tracing
+func (l *TracingRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	ctx := req.Context()
+
+	span := sentry.StartTransaction(ctx, fmt.Sprintf("%s %s", req.Method, req.URL.Path))
```

Comment:
Should this round tripper contain the "hub" logic here? Might be cases where this roundtripper is used on a client that we use from outside of a request flow, so I'm thinking if it would make sense to have that in place here or not. Or is the whole "hub" mechanism relevant for keeping track of requests that touch multiple services? 🤔

Reply:
I've added the hub logic.